packages/react/src/combobox/ComboboxBase/ComboboxBase.tsx

```diff
@@ -101,6 +105,14 @@ export function ComboboxBase<Option extends DefaultComboboxOption>({
 		paddingRight: hasBothButtons ? '5rem' : '3rem',
 	};
 
+	const itemLabels = useMemo(
```

Comment:
Memoising the split up label characters for matching improves the speed a good chunk